Makefile

```diff
@@ -0,0 +1,3 @@
+test:
+	@echo "Testing ..."
+	@go clean -testcache && go test ./...
```

Comment:
Git warns about newline. Please add a newline at the end.

Reply:
done

Reply:
A couple of more changes here. Please add the following flags to go test

1. `-v` for verbosity
2. `-race` for data race detection
3. `-coverprofile cover.out` for coverage profile

With `cover.out`, we can get the coverage. Please add this in next line.
```
go tool cover -func cover.out
```
Also, `go test` needn't be hid from stdout. You may keep `go clean` in `@` and take `go test` command to next line.

Reply:
done